src/components/AddMovie.tsx

```diff
@@ -1,30 +1,178 @@
 import React from 'react';
 import styled from 'styled-components';
+import { Formik, Form, Field } from 'formik';
+import * as Yup from 'yup';
+import { MovieData } from '../model';
+import { selectSelectedMovie } from '../store/moviesSlice';
+import { useAppSelector } from '../store/hooks';
+import { GENRES } from './NavBar';
 
 interface AddUpdateMovieProps {
-  onAddMovie?: any;
-  onUpdateMovie?: any;
+  onAddMovie?: (isDataUpdated: boolean, data: MovieData) => void;
+  onUpdateMovie?: (isDataUpdated: boolean, data: MovieData) => void;
 }
 
-const AddUpdateMovieWrapper = styled.div``;
+const AddUpdateMovieWrapper = styled.div`
+  padding: 10px;
+`;
+
+const ItemWrapper = styled.div`
+  display: flex;
+  justify-content: space-between;
+`;
+
+const ErrorMessageWrapper = styled.span`
+  color: red;
+`;
 
 const AddUpdateMovie = (props: AddUpdateMovieProps) => {
-  const onUpdateClick = () => {
-    props.onUpdateMovie(true);
+  const selectedMovie = useAppSelector(selectSelectedMovie);
+
+  const SignupSchema = Yup.object().shape({
+    title: Yup.string().required('Required'),
+    release_date: Yup.string()
+      .required('Required')
+      .matches(/\d{4}-\d{2}-\d{2}/, 'The correct format is 2022-12-12'),
+    rating: Yup.number(),
+    genres: Yup.array().of(Yup.string()).min(1, 'Required'),
+    runtime: Yup.number().required('Required'),
+    overview: Yup.string().required('Required'),
+    poster_path: Yup.string().required('Required'),
+  });
+
+  const onUpdateMovie = (data: MovieData) => {
+    props.onUpdateMovie(true, data);
   };
 
-  const onAddMovie = () => {
-    props.onAddMovie(true);
+  const onAddMovie = (data: MovieData) => {
+    props.onAddMovie(true, data);
   };
 
   return (
     <AddUpdateMovieWrapper>
       <h3>{props.onUpdateMovie ? 'UPDATE MOVIE' : 'ADD MOVIE'}</h3>
-      {props.onUpdateMovie ? (
-        <button onClick={onUpdateClick}>UPDATE</button>
-      ) : (
-        <button onClick={onAddMovie}>ADD</button>
-      )}
+
+      <Formik
+        initialValues={
+          selectedMovie
+            ? {
+              title: selectedMovie.title,
+              release_date: selectedMovie.release_date,
+              vote_average: selectedMovie.vote_average,
+              genres: selectedMovie.genres,
+              overview: selectedMovie.overview,
+              poster_path: selectedMovie.poster_path,
+              runtime: selectedMovie.runtime,
+            }
+            : {
+              title: '',
+              release_date: '',
+              vote_average: 0,
+              genres: [],
+              overview: '',
+              poster_path: '',
+              runtime: 0,
+            }
+        }
+        validationSchema={SignupSchema}
+        onSubmit={(values) => {
+          if (props.onUpdateMovie) {
+            onUpdateMovie(values);
+          } else {
+            onAddMovie(values);
+          }
```

Comment:
If you handle updating / adding here, you couple the logic in this component. Could you pass one callback onSumit and change the callback in the parent component?

Reply:
Fixed.